armi/interfaces.py

```diff
@@ -77,6 +80,117 @@ class STACK_ORDER:  # pylint: disable=invalid-name, too-few-public-methods
+
+        Notes
+        -----
+        This checks the type of the value against ``_SUPPORTED_TYPES`` before storing.
+        """
+        if type(val) not in self._SUPPORTED_TYPES:
+            raise TypeError(
+                f"{val} supplied has type {type(val)} which is not supported in {self}. "
+                f"Supported types: {self._SUPPORTED_TYPES}"
+            )
+        self._previousIterationValue = val
+
+    def isConverged(self, val):
+        """
+        Return True if the convergence criteria between the current and previous iteration values are met.
+
+        Notes
+        -----
+        On convergence, this class is automatically reset to its initial condition to avoid retaining
+        or holding a stale state. Calling this method will increment a counter that when exceeded will
+        clear the state. A warning will be reported if the state is cleared prior to the convergence
+        criteria being met.
+
+        Raises
+        ------
+        ValueError
+            If the previous iteration value has not been assigned. The ``storePreviousIterationValue`` method
+            must be called first.
+
+        TypeError
+            If the type of the current value provided is not the same as the previous iteration value.
+        """
+        if self._previousIterationValue is None:
+            raise ValueError(
+                f"Cannot check convergence of {self} with no previous iteration value set. "
+                f"Set using `storePreviousIterationValue` first."
+            )
+
+        if not isinstance(val, type(self._previousIterationValue)):
+            raise TypeError(
+                f"The current value {val} is not the same type as the previous "
+                f"iteration value of {self._previousIterationValue}"
+            )
+
+        previous = self._previousIterationValue
+
+        # Check convergence for integer or float values.
+        if isinstance(val, int) or isinstance(val, float):
+            self.eps = abs(val - previous)
+
+        # Convert list values to a numpy array.
+        if isinstance(val, list):
+            val = numpy.array(val)
+            previous = numpy.array(previous)
+
+        # Check convergence for numpy array values.
+        if isinstance(val, numpy.ndarray):
+            epsVec = []
+            for old, new in zip(previous, val):
+                epsVec.append(norm(numpy.subtract(old, new), ord=2))
+            self.eps = norm(epsVec, ord=numpy.inf)
+
+        # Check convergence and if convergence is satisfied then reset the state of this
+        # object back to its originally defined state by calling __init__(...)
+        converged = self.eps < self.tolerance
+        if converged:
+            self.__init__(self.param, self.tolerance, self.maxIters)
+        else:
+            self._numIters += 1
+            if self._numIters == self.maxIters:
+                self.__init__(self.param, self.tolerance, self.maxIters)
```

Comment:
Need to add a runLog warning if the number iterations equals the maximum and this is reset.

Reply:
https://github.com/terrapower/armi/pull/1033/commits/ceb13d8d90dc248b66f476ceea293b6e93d16100